src/net/java/sip/communicator/impl/protocol/jabber/ChatRoomJabberImpl.java

```diff
@@ -1028,6 +1028,40 @@ public void sendMessage(Message message)
          }
     }
 
+    /**
+     * Sends the <tt>message</tt> with the json-message extension to the
+     * destination indicated by the <tt>to</tt> contact.
+     *
+     * @param message the <tt>Message</tt> to send.
+     * @param json the json message to be sent.
+     * @throws OperationFailedException if sending the message fails for some
+     * reason.
+     */
+    public void sendJsonMessage(Message message, String json)
+            throws OperationFailedException
+    {
+        try
+        {
+            assertConnected();
+            org.jivesoftware.smack.packet.Message msg =
+                    new org.jivesoftware.smack.packet.Message();
+            msg.setType(org.jivesoftware.smack.packet.Message.Type.groupchat);
+            msg.setBody(message.getContent());
```

Comment:
Do we want to set an empty body? @damencho pointed out that if we include a body, the json-message will be "stored" in the chat history and replayed when someone else joins. Have you tried if it works without a body?

Reply:
Though we set a null body, the sent Message stanza does not have a body. I'll try without it once.